cirq-superstaq/cirq_superstaq/validation.py

```diff
@@ -4,10 +4,46 @@
 
 import cirq
 
+SUPPORTED_QID_TYPES = (
+    cirq.LineQubit,
+    cirq.LineQid,
+    cirq.GridQubit,
+    cirq.GridQid,
+    cirq.NamedQubit,
+    cirq.NamedQid,
+)
+
+
+def validate_qubit_types(circuits: object) -> None:
+    """Verifies that `circuits` consists of valid qubit types only.
+
+    Args:
+        circuits: The input circuit(s) to validate.
+
+    Raises:
+        TypeError: If an invalid qubit type is found in `circuits`.
+    """
+    circuits_to_check = [circuits] if isinstance(circuits, cirq.Circuit) else circuits
+    assert isinstance(circuits_to_check, Sequence)
+
+    all_qubits_present: set[cirq.Qid] = set()
+    for circuit in circuits_to_check:
+        all_qubits_present.update(circuit.all_qubits())
+
+    if not all(isinstance(q, SUPPORTED_QID_TYPES) for q in all_qubits_present):
+        raise TypeError(
+            "Input circuit(s) contain unsupported qubit types. Valid qubit types are: "
```

Comment:
```suggestion
            f"Input circuit(s) contain unsupported qubit types: {', '.join(set(type(q) for q in all_qubits_present) - SUPPORTED_QID_TYPES)}. Valid qubit types are: "
```

Reply:
This crossed my mind but I forgot to come back to it. Adapted your suggestion https://github.com/Infleqtion/client-superstaq/pull/1050/commits/3e04c18ff662916e945a889c844373a6cb349bca!